crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_structure.rs

```diff
@@ -67,23 +57,22 @@ pub(crate) const fn row_and_column_from_index(index: usize) -> (usize, usize) {
 
 /// Returns the index of data where the (row, column) belongs.
 #[allow(dead_code)]
-pub(crate) const fn index_from_row_and_column(row: usize, column: usize) -> usize {
+pub(crate) fn index_from_row_and_column(row: usize, column: usize) -> Option<usize> {
     let width_of_row = full_width_of_row(row);
-
-    // Return no_index if the (row, column) is not in dynamic Dory structure.
-    if column >= width_of_row || (row == 1 && column == 0) {
-        return no_index();
-    }
-
-    width_of_row * (row - width_of_row / 2) + column
+    (column < width_of_row && (row, column) != (1, 0))
+        .then_some(width_of_row * (row - width_of_row / 2) + column)
 }
 
-/// Returns a matrix size that can hold the given number of data points being committed with respect to an offset.
+/// Returns a matrix size, (height, width), that can hold the given number of data points being committed with respect to an offset.
 #[allow(dead_code)]
 pub(crate) const fn matrix_size(data_len: usize, offset: usize) -> (usize, usize) {
+    if data_len == 0 {
+        return (0, 0);
+    }
+
     let (last_row, _) = row_and_column_from_index(offset + data_len - 1);
     let width_of_last_row = full_width_of_row(last_row);
-    (width_of_last_row, last_row + 1)
+    (last_row + 1, width_of_last_row)
```

Comment:
I didn't mean to make you switch it. But, (height, width) feels better, so I'm not complaining.